ansible/roles/sshkey-host-access/tasks/main.yml

```diff
@@ -0,0 +1,73 @@
+---
+
+- name: ensure the existence of the user directory
+  file:
+    path: "{{ KEY_LOC_PATH }}"
+    owner: "{{ KEY_LOC_OWNER }}"
+    group: "{{ KEY_LOC_GROUP }}"
+    mode: 0700
+    state: directory
+  delegate_to: "{{ play_on }}"
+
+- block:
+    - name: check the existence of a given ssh key by name
+      stat:
+        path: "{{ KEY_LOC_PATH }}/{{ ATMOUSERNAME }}{{ KEY_LOC_PREFIX }}_default"
+      register: go_key
+
+    - name: generate the keypair
+      command: >
+        /usr/bin/ssh-keygen -N '' -t rsa -b 2048 -f {{ KEY_LOC_PATH }}/{{ ATMOUSERNAME }}{{ KEY_LOC_PREFIX }}_default -C {{ ATMOUSERNAME }}
+      when: go_key.stat.exists == False
+
+    - name: set the permissions on the private key file
+      file:
+        path: "{{ KEY_LOC_PATH }}/{{ ATMOUSERNAME }}{{ KEY_LOC_PREFIX }}_default"
```

Comment:
This value `"{{ KEY_LOC_PATH }}/{{ ATMOUSERNAME }}{{ KEY_LOC_PREFIX }}_default"` is replicated many times, which is hard to read and prone to typos. Can value be moved to `vars/defaults.yml`?